test/langtools/tools/javac/annotations/crashOnUnknownAttr/CrashOnUnknownTargetTypeTest.java

```diff
@@ -0,0 +1,11 @@
+/*
+ * @test /nodynamiccopyright/
+ * @summary compiler is crashing with AssertionError for annotations with unknown target type
+ * @bug 8296010
+ * @build A
+ * @compile/fail/ref=CrashOnUnknownTargetTypeTest.out -XDrawDiagnostics CrashOnUnknownTargetTypeTest.java
+ */
+
+public class CrashOnUnknownTargetTypeTest {
+    @A Object o;
```

Comment:
I wonder if it's worth testing the case where the annotation appears on a method as well?  We observed crashes before due to that placement as well.  It's possible the relevant code paths are shared so an additional test is unnecessary

Reply:
thanks for your feedback. Yes we could do that for completeness, will take a look at this. Side: did you file any bug on the issue you found? Could you please try this fix on your code base and see if the issue gets fixed? Thanks!

Reply:
I originally reported https://bugs.openjdk.org/browse/JDK-8295314, which is a bug on JDK 8 triggered when an annotation targets `MODULE`.  Then, through discussions with @cushon we realized the same issue is present on the latest JDK when the annotation target is unknown, at which point he reported https://bugs.openjdk.org/browse/JDK-8296010.

Trying this patch on my own example may be a bit tricky, since that was related to JDK 8 and I don't have a lot of cycles to look into it right now.  If I get time to try something out, I will report back here.

Reply:
I see, thanks, yes https://bugs.openjdk.org/browse/JDK-8295314 is definitely a duplicate of this one